models/QtiRunnerInitDataBuilder.php

```diff
@@ -303,4 +314,28 @@ private function getResponseCountsFromState(array $itemState): int
 
         return $responsesCount;
     }
+
+    /**
+     * @throws MissingOutcomeDeclarationMaxScore
+     */
+    private function getMaxScore($outcomes): float
+    {
+        foreach ($outcomes as $outcome) {
+            if ($outcome[self::OUTCOME_DECLARATION_IDENTIFIER] === self::OUTCOME_DECLARATION_MAXSCORE) {
+                return (float) $outcome[self::OUTCOME_DECLARATION_DEFAULT_VALUE];
+            }
+        }
+
+        throw new MissingOutcomeDeclarationMaxScore();
+    }
+
+    /**
+     * @throws DataItemNotValidException
+     */
+    private function validateItemData($itemData): void
+    {
+        if (!isset($itemData['data'], $itemData['data']['attributes'], $itemData['data']['outcomes'])) {
```

Comment:
No need to check the first condition.
```suggestion
        if (!isset($itemData['data']['attributes'], $itemData['data']['outcomes'])) {
```